### DISCUSSION TITLE
How to connect external sensors to the Crazyflie 2.1 and do calibration?

### DISCUSSION THREAD

**Discussion by Jashu003** [bitcraze/discussions#1311] [Q&A]
Hi everyone,

I have a question regarding integrating an LDR (Light Dependent Resistor) sensor with the Crazyflie 2.1, and I believe the answer would benefit many others in the community as well.

My query is as follows: Can we directly connect an LDR sensor to the Crazyflie 2.1 through its Analog pins? If so, what steps are involved, and do we need to modify the code running on the Crazyflie after connecting the LDR?

Additionally, if anyone could recommend a suitable LDR sensor for this application, it would be greatly appreciated.

Looking forward to your insights and recommendations!

Regards,
Jashwanth

**Comment by tobbeanton** [comment]
Integrating an LDR should not be to complicated. I would use a [prototyping deck](https://www.bitcraze.io/products/prototyping-deck/) as a base. When it comes to what LDR to select I think you would have to list more requirements or describe your use-case. The Crazyflie [deck expansion ports](https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#expansion-port-pinout) has analog input which should be in the span 0-3V.  

When you have soldered hardware you need to write a driver and you could use the [deck driver howto](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/howto/) as a starting point.
